### PR TITLE
Fix validation for SharedContent

### PR DESCRIPTION
**Pull request by tomtollafield**

### CODE REVIEW

File: DFC.App.Pages.Data/Helpers/ContentTypeHelper.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
+﻿using DFC.App.Pages.Data.Enums;
+using System;
+using System.IO;
+
+namespace DFC.App.Pages.Data.Helpers
+{
+    public static class ContentTypeHelper
+    {
+        public static EventContentType GetContentTypeFromUrl(this Uri url)
```
Comment (lazcool) [inline DFC.App.Pages.Data/Helpers/ContentTypeHelper.cs:9]: Looks like we need to add content type to the event's data 😃.

File: DFC.App.Pages.Data.UnitTests/HelperTests/ContentTypeHelperTests.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+using DFC.App.Pages.Data.Helpers;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Xunit;
+
+namespace DFC.App.Pages.Data.UnitTests.HelperTests
+{
+    [Trait("Category", "ContentTypeHelper Unit Tests")]
+    public class ContentTypeHelperTests
```
Comment (lazcool) [inline DFC.App.Pages.Data.UnitTests/HelperTests/ContentTypeHelperTests.cs:11]: Might be worth an unit test for when the url doesn't contain the scheme or host.

File: DFC.App.Pages.Data/Helpers/ContentTypeHelper.cs (hunk 1 above)
Comment (tomtollafield) [reply DFC.App.Pages.Data/Helpers/ContentTypeHelper.cs:9]: That would be handy

File: DFC.App.Pages.Services.CacheContentService/WebhooksService.cs [hunk 3]
```diff
@@ -246,10 +247,17 @@ public bool RemoveContentItem(Guid contentItemId, List<ContentItemModel>? items)
...
-        public bool TryValidateModel(ContentPageModel? contentPageModel)
+        public bool TryValidateModel(ContentPageModel? contentPageModel, Uri contentUrl)
         {
             _ = contentPageModel ?? throw new ArgumentNullException(nameof(contentPageModel));
 
+            var eventContentType = contentUrl.GetContentTypeFromUrl();
+
+            if (eventContentType == EventContentType.SharedContent)
+            {
+                contentPageModel.CanonicalName = "sharedcontent";
```
Comment (lazcool) [inline DFC.App.Pages.Services.CacheContentService/WebhooksService.cs:258]: Is it OK to have the same canonical name for all shared content items?

**Pull request merged**